cache/cache.go

```diff
@@ -111,23 +101,18 @@ func (fs FSCache) getLayer(decompressedBucket, layerBucket *bolt.Bucket, layerID
 	return l, nil
 }
 
-func (fs FSCache) PutLayer(layerID, decompressedLayerID string, layerInfo types.LayerInfo) error {
+func (fs FSCache) PutLayer(diffID string, layerInfo types.LayerInfo) error {
+	if _, err := v1.NewHash(diffID); err != nil {
+		return xerrors.Errorf("invalid diffID (%s): %w", diffID, err)
+	}
+
 	b, err := json.Marshal(layerInfo)
 	if err != nil {
 		return err
 	}
 	err = fs.db.Update(func(tx *bolt.Tx) error {
-		// store a mapping from a layer ID to a decompressed layer ID.
-		if layerID != decompressedLayerID {
-			decompressedBucket := tx.Bucket([]byte(decompressedDigestBucket))
-			err := decompressedBucket.Put([]byte(layerID), []byte(decompressedLayerID))
-			if err != nil {
-				return xerrors.Errorf("unable to store a pair of compressed/decompressed layer IDs: %w", err)
-			}
-		}
-
 		layerBucket := tx.Bucket([]byte(layerBucket))
-		err = layerBucket.Put([]byte(decompressedLayerID), b)
+		err = layerBucket.Put([]byte(diffID), b)
```

Comment:
ditto, potential for panic.

Reply:
https://github.com/aquasecurity/fanal/pull/96#discussion_r405443752